controllers/secretproviderclasspodstatus_controller.go

```diff
@@ -227,18 +226,46 @@ func (r *SecretProviderClassPodStatusReconciler) Reconcile(req ctrl.Request) (ct
 		return ctrl.Result{}, nil
 	}
 
-	// podObjectReference is an object reference to the pod that spc pod status
-	// is created for. The object reference is created with minimal required fields
-	// name, namespace and UID. By doing this we can skip an additional client call
-	// to fetch the pod object
-	podObjectReference, err := getPodObjectReference(spcPodStatus)
-	if err != nil {
-		logger.Errorf("failed to get pod object reference, error: %+v", err)
+	// Obtain the full pod metadata. An object reference is needed for sending
+	// events and the UID is helpful for validating the SPCPS TargetPath.
+	pod := &v1.Pod{}
+	if err := r.reader.Get(ctx, client.ObjectKey{Namespace: req.Namespace, Name: spcPodStatus.Status.PodName}, pod); err != nil {
+		logger.Errorf("failed to get pod %s/%s, err: %+v", req.Namespace, spcPodStatus.Status.PodName, err)
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	// determine which pod volume this is associated with
+	var podVol *v1.Volume
+	for i, vol := range pod.Spec.Volumes {
```

Comment:
Not a blocker for this PR but I think we can make this a common function which returns volume based on spc name. We can then reuse it in the rotation reconciler for getting the `nodePublishSecretRef` name.

Reply:
Yeah I just was having a hard time naming a new package for it. Maybe a new package under `/pkg/util/k8sutil`, but `util` packages are generally discouraged. Happy to move it though!

Reply:
yeah, `/pkg/util/k8sutil` makes the most sense here. We could do that.

Reply:
Pushed a latest commit with the new package